### PR TITLE
[EPMEDU-1052]: Open Search page

### PR DESCRIPTION
**Pull request by VEZE**
- Implemented AnimalTypeHorizontalPager
- Implemented LazyColumn AnimalType content: scrollable, searchable, collapsible.
- Implemented EmptyState.
- Implemented SearchView.
- Added base implementation of local search, modified repositories for receiving list of animals by type.

https://user-images.githubusercontent.com/11144164/226437552-e8f88759-f8c5-49f4-86cb-98e9d7df3447.mp4

### CODE REVIEW

File: library/foundation/src/main/java/com/epmedu/animeal/foundation/tabs/AnimealPagerTabRow.kt [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+}
+
+@Composable
+private fun AnimealPagerTabIndicator(
+    modifier: Modifier = Modifier,
+    color: Color = MaterialTheme.colors.primary
+) {
+    Spacer(
+        modifier
+            .padding(start = 30.dp, end = 30.dp)
```
Comment (Fileek) [inline library/foundation/src/main/java/com/epmedu/animeal/foundation/tabs/AnimealPagerTabRow.kt]: Without these paddings, the indicator would be more consistent with the design.

File: feature/tabsflow/search/src/main/java/com/epmedu/animeal/tabs/search/presentation/search/SearchScreen.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package com.epmedu.animeal.tabs.search.presentation.search
```
Comment (Fileek) [inline feature/tabsflow/search/src/main/java/com/epmedu/animeal/tabs/search/presentation/search/SearchScreen.kt]: The last `search` subpackage is redundant

File: feature/tabsflow/search/src/main/java/com/epmedu/animeal/tabs/search/presentation/search/SearchView.kt [hunk 3]
```diff
@@ -0,0 +1,136 @@
+package com.epmedu.animeal.tabs.search.presentation.search
```
Comment (Fileek) [inline feature/tabsflow/search/src/main/java/com/epmedu/animeal/tabs/search/presentation/search/SearchView.kt]: Please move `SearchView` to `presentation.ui` subpackage

File: feature/tabsflow/search/src/main/java/com/epmedu/animeal/tabs/search/presentation/search/SearchScreenViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,134 @@
+package com.epmedu.animeal.tabs.search.presentation.search
```
Comment (Fileek) [inline feature/tabsflow/search/src/main/java/com/epmedu/animeal/tabs/search/presentation/search/SearchScreenViewModel.kt]: Please move `SearchScreenViewModel` to `presentation.viewmodel` subpackage

File: feature/tabsflow/search/src/main/java/com/epmedu/animeal/tabs/search/presentation/search/SearchState.kt [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package com.epmedu.animeal.tabs.search.presentation.search
```
Comment (Fileek) [inline feature/tabsflow/search/src/main/java/com/epmedu/animeal/tabs/search/presentation/search/SearchState.kt]: Please move `SearchState` to `presentation.viewmodel` subpackage

File: feature/tabsflow/search/src/main/java/com/epmedu/animeal/tabs/search/presentation/search/SearchView.kt [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+                color = CustomColor.TrolleyGrey
+            )
+        )
+    }
+}
+
+@AnimealPreview
+@Composable
+private fun SearchViewPreview() {
+    SearchView()
```
Comment (Fileek) [inline feature/tabsflow/search/src/main/java/com/epmedu/animeal/tabs/search/presentation/search/SearchView.kt]: Don't forget to wrap your previews in `AnimealTheme`

**Review by Fileek — CHANGES_REQUESTED** [review]
Good job!
I merged the develop branch to your PR so we can test this feature and left some comments about the design and packages. Please fix them and I will merge this PR.

File: library/foundation/src/main/java/com/epmedu/animeal/foundation/tabs/AnimealPagerTabRow.kt (hunk 1 above)
Comment (VEZE) [reply library/foundation/src/main/java/com/epmedu/animeal/foundation/tabs/AnimealPagerTabRow.kt]: Removed

File: feature/tabsflow/search/src/main/java/com/epmedu/animeal/tabs/search/presentation/search/SearchScreenViewModel.kt (hunk 4 above)
Comment (VEZE) [reply feature/tabsflow/search/src/main/java/com/epmedu/animeal/tabs/search/presentation/search/SearchScreenViewModel.kt]: moved

File: feature/tabsflow/search/src/main/java/com/epmedu/animeal/tabs/search/presentation/search/SearchState.kt (hunk 5 above)
Comment (VEZE) [reply feature/tabsflow/search/src/main/java/com/epmedu/animeal/tabs/search/presentation/search/SearchState.kt]: moved

File: feature/tabsflow/search/src/main/java/com/epmedu/animeal/tabs/search/presentation/search/SearchView.kt (hunk 6 above)
Comment (VEZE) [reply feature/tabsflow/search/src/main/java/com/epmedu/animeal/tabs/search/presentation/search/SearchView.kt]: wrapped

File: feature/tabsflow/search/src/main/java/com/epmedu/animeal/tabs/search/presentation/search/SearchScreen.kt (hunk 2 above)
Comment (VEZE) [reply feature/tabsflow/search/src/main/java/com/epmedu/animeal/tabs/search/presentation/search/SearchScreen.kt]: removed

File: feature/tabsflow/search/src/main/java/com/epmedu/animeal/tabs/search/presentation/search/SearchView.kt (hunk 3 above)
Comment (VEZE) [reply feature/tabsflow/search/src/main/java/com/epmedu/animeal/tabs/search/presentation/search/SearchView.kt]: moved to foundation

**Review by Fileek — APPROVED** [review]

**Pull request merged**